### PR TITLE
Replay logs to debug

### PR DESCRIPTION
**Pull request by colegleason**

### CODE REVIEW

File: ja.lproj/Localizable.strings [hunk 1]
```diff
@@ -120,6 +120,9 @@
 /* Button to start navigation */
 "startNavigationButton" = "ナビ開始";
 
+/* Button to start log replay */
+"startReplayButton" = "Start Log Replay";
```
Comment (colegleason) [inline ja.lproj/Localizable.strings]: needs to be changed

File: NavCog/Model/StateMachine/NavMachine.m [hunk 2]
```diff
@@ -355,9 +369,171 @@ - (void)startNavigationOnTopoMap:(TopoMap *)topoMap fromNodeWithName:(NSString *
...
+        }
+    }
+    
+    return logList;
+}
+
 - (void)stopNavigation {
     [self stopAudio];
-    [_beaconManager stopRangingBeaconsInRegion:_beaconRegion];
+    if (!(_logReplay)) {
```
Comment (colegleason) [inline NavCog/Model/StateMachine/NavMachine.m:525]: Why do we only stop if not replaying logs?

File: NavCog/Model/StateMachine/NavMachine.m [hunk 3]
```diff
@@ -382,11 +558,16 @@ - (void)announceAccessibilityInfo {
     }
 }
 
+//TODO: this even used?
```
Comment (colegleason) [inline NavCog/Model/StateMachine/NavMachine.m:553]: I don't believe so. In XCode you should be able to see the call hierarchy.

File: NavCog/Model/StateMachine/NavMachine.m [hunk 4]
```diff
@@ -424,10 +609,12 @@ - (void)locationManager:(CLLocationManager *)manager didRangeBeacons:(NSArray *)
                     if (_currentState == nil) {
                         [_delegate navigationFinished];
                         [NavNotificationSpeaker speakWithCustomizedSpeed:NSLocalizedString(@"arrived", @"Spoken when you arrive at a destination")];
-                        [_beaconManager stopRangingBeaconsInRegion:_beaconRegion];
+                        if (!(_logReplay)) {
```
Comment (colegleason) [inline NavCog/Model/StateMachine/NavMachine.m]: see above

File: NavCog/Model/StateMachine/NavMachine.m [hunk 5]
```diff
@@ -424,10 +609,12 @@ - (void)locationManager:(CLLocationManager *)manager didRangeBeacons:(NSArray *)
...
+                            [_beaconManager stopRangingBeaconsInRegion:_beaconRegion];
+                        }
                         [_topoMap cleanTmpNodeAndEdges];
                     } else if (_currentState.type == STATE_TYPE_WALKING) {
-//                        if (ABS(_curOri - _currentState.ori) > 15) {
+                        //                        if (ABS(_curOri - _currentState.ori) > 15) {
```
Comment (colegleason) [inline NavCog/Model/StateMachine/NavMachine.m:609]: please remove whitespace

File: NavCog/Model/StateMachine/NavMachine.m (hunk 2 above)
Comment (colegleason) [inline NavCog/Model/StateMachine/NavMachine.m:525]: Maybe we write a wrapper around startRanging and stopRanging to handle the `_logReplay` check.

**Comment by colegleason** [comment]
@loopRW is this PR still relevant?

**Comment by loopRW** [comment]
It is not, due to rebasing issues i merged the replayer manually. Current Master has stable replayer already

**Comment by colegleason** [comment]
Ok, great.

**Pull request closed without merging**